### PR TITLE
chore(rl): standarize redis configuration

### PR DESCRIPTION
**Pull request by nowNick**
### Summary

Rate Limiting right now has new config structure that reuses common redis connection configuration.

Many plugins differ in the way Redis is configured. The goal of this PR is to introduce a standard Redis config schema that could be shared across plugins.

### Other PRs
This is the second in the chains of PRs that standardize Redis configuration.

1. ACME: https://github.com/Kong/kong/pull/12300
2. RateLimiting: (this PR)
3. Response-RateLimiting: https://github.com/Kong/kong/pull/12302

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/README.md)
- [x] N/A (docs automatically generated from plugin schema description) ~There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE~ 

### Issue reference

KAG-3388

### CODE REVIEW

**Review by jschmid1 — APPROVED** [review]
looks great 🚀

File: spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua [hunk 1]
```diff
@@ -362,6 +362,51 @@ describe("CP/DP config compat transformations #" .. strategy, function()
...
+              redis = {
+                  host = "localhost",
+                  port = 57198,
+                  username = "test",
+                  password = "secret",
+                  database = 2,
+                  timeout = 1100,
+                  ssl = true,
+                  ssl_verify = true,
+                  server_name = "example.test"
```
Comment (locao) [inline spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua:385]: I am a simple man, I see a `.test` hostname, I love the code ❤️

**Review by locao — APPROVED** [review]
Good work!

**Comment by locao** [comment]
@nowNick I merged the other two PRs, but now we have conflicts in this one. Can you check, please?

**Pull request merged**